src/fibsem_tools/io/fibsem_h5.py

```diff
@@ -0,0 +1,314 @@
+    dat_filenames: Sequence[str], h5_dirname: str = None, **kwargs
+):
+    """
+    Create aggregate FIBSEM HDF5 File from .dat filenames
+
+    :param dat_filenames: A sequence of dat filenames
+    :param h5_filename: Name of HDF5 FILE. defaults to same directory as dat_filenames[0]
+    :type h5_filename: str, optional
+
+    :returns: h5_filename
+    """
+    prefix_name = ""
+    dataset_names = []
+    raw_headers = []
+
+    # Parse filenames of the form prefix_0_1_2.dat
+    r_dat_filename = re.compile(r"(.*)_(\d+-\d+-\d+).dat")
+    for dat_filename in dat_filenames:
+        dat_filename_base = os.path.basename(dat_filename)
+        m = re.fullmatch(r_dat_filename, dat_filename_base)
+        dataset_names.append(m[2])
+        raw_headers.append(_extract_raw_header(dat_filename))
+        if prefix_name == "":
+            prefix_name = m[1]
+        else:
+            # Ensure that all dat files have the same prefix
+            assert prefix_name == m[1], "All arguments must share a common prefix"
+
+    data = read_fibsem(dat_filenames)
+    if h5_dirname is None:
+        h5_dirname = os.path.dirname(dat_filenames[0])
+    h5_filename = os.path.join(h5_dirname, prefix_name + ".h5")
+
+    return create_aggregate_fibsem_h5_file(
+        h5_filename, dataset_names, data, raw_headers, **kwargs
+    )
+
+
+def create_default_h5_files(source_data, **kwargs):
+    """
+    Create HDF5 files with default settings
+
+    This is a high-level convenience function.
+
+    :param source_data: Source data. .dat filenames, a directory name, or a glob
+    :param chunk: Chunk size as tuple (Y, X)
+
+    :return: list of HDF5 filenames or `None`
+    """
+    if os.path.isdir(source_data):
+        source_data = os.path.join(source_data, "*.dat")
+
+    if "*" in source_data:
+        source_data = glob.glob(source_data)
+
+    if not isinstance(source_data, list):
+        source_data = [source_data]
+
+    if isinstance(source_data[0], str):
+        # If str is given, assume these are .dat filenames
+
+        # Group dat filenames by prefix, and put them into prefix specific HDF5 files
+        r_dat_filename = re.compile(r"(.*)_(\d+-\d+-\d+).dat")
+        prefixes = [re.fullmatch(r_dat_filename, fn)[1] for fn in source_data]
+        uniq_prefixes, ind_prefixes = np.unique(prefixes, return_inverse=True)
+        h5_files = []
+        for i, prefix in enumerate(uniq_prefixes):
+            dat_filename_group = [
+                d for d, ip in zip(source_data, ind_prefixes) if ip == i
+            ]
+            h5f = create_aggregate_fibsem_h5_file_from_dat_filenames(
+                dat_filename_group, **kwargs
+            )
+            h5_files.append(h5f)
+
+        return h5_files
+    elif instanceof(source_data[0], FIBSEMData):
+        # TODO handle more situations
+        raise NotImplemented
+
+    return None
+
+
+def create_zstd_h5_files(source_data, aggression=1, **kwargs):
```

Comment:
this function and the subsequent function don't really need to exist, since their only difference is which compressor is being used, and `h5py` already provides that flexibility. we can just have 1 function that wraps the h5py api and let the user pick the compressor they want